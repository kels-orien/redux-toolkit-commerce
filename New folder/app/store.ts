import { configureStore, getDefaultMiddleware } from "@reduxjs/toolkit"
import { apiSlice } from "../features/products/products-api-slice"
import cartReducer  from "../features/cart/cart-slice"


export const store =  configureStore({
    reducer: {
        cart: cartReducer,
        [apiSlice.reducerPath]: apiSlice.reducer
    },
    middleware: (getDefaultMiddleware) => {
        return getDefaultMiddleware().concat(apiSlice.middleware)
    }   
})

export type AppDispatch =  typeof store.dispatch
export type RootState = ReturnType<typeof store.getState>