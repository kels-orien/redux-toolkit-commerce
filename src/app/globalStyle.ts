import { createGlobalStyle } from 'styled-components';




export const GlobalStyle = createGlobalStyle`
*,::after,::before{
    background-repeat:no-repeat;
    box-sizing:inherit
}
::after,::before{
    text-decoration:inherit;
    vertical-align:inherit
}
@font-face {
    font-family: NewsGothicBTW01-Roman;
    font-weight: 400;
    src: url(../fonts/NewsGothicBTW01Roman) format('woff'),
    url(../fonts/NewsGothicBTW01Roman2) format('woff2');
    font-style: normal;
}
html{
    box-sizing:border-box;
    cursor:default;
    -ms-text-size-adjust:100%;
    -webkit-text-size-adjust:100%
}
article,aside,footer,header,nav,section{
    display:block
}
body{
    margin:0;
    height: 100%;
}
h1{
    font-size:2em;
    margin:.67em 0
}
figcaption,figure,main{
    display:block
}
figure{
    margin:1em 40px
}
hr{
    box-sizing:content-box;
    height:0;
    overflow:visible
}
nav ol,nav ul{
    list-style:none
}
pre{
    font-family:monospace,monospace;
    font-size:1em
}
a{
    text-decoration:none;
    color:inherit;
    background-color:transparent;
    -webkit-text-decoration-skip:objects
}
abbr[title]{
    border-bottom:none;
    text-decoration:underline;
    text-decoration:underline dotted
}
b,strong{
    font-weight:inherit
}
b,strong{
    font-weight:bolder
}
code,kbd,samp{
    font-family:monospace,monospace;
    font-size:1em
}
dfn{
    font-style:italic
}
mark{
    background-color:#ff0;
    color:#000
}
small{
    font-size:80%
}
sub,sup{
    font-size:75%;
    line-height:0;
    position:relative;
    vertical-align:baseline
}
sub{
    bottom:-.25em
}
sup{
    top:-.5em
}
::-moz-selection{
    background-color:#b3d4fc;
    color:#000;
    text-shadow:none
}
::selection{
    background-color:#b3d4fc;
    color:#000;
    text-shadow:none
}
h2 {
    display: block;
    font-size: 1.5em;
    margin-block-start: 0em;
    margin-block-end: 0em;
    margin-inline-start: 0px;
    margin-inline-end: 0px;
    font-weight: bold;
}
audio,canvas,iframe,img, video{
    vertical-align:middle
}
audio,video{
    display:inline-block
}
audio:not([controls]){
    display:none;
    height:0
}
img{
    border-style:none
}
svg:not(:root){
    overflow:hidden
}
table{
    border-collapse:collapse
}
input[type=number]::-webkit-inner-spin-button, 
input[type=number]::-webkit-outer-spin-button { 
  -webkit-appearance: none; 
}

input[type=number] {
  -moz-appearance: textfield;
}
button[type=submit]{border:none}
button {
    border: none;
 }
 button: hover {
    border: none;
 }
 li button span: focus {
    border: none;
 }
 input: focus {
     outline:none
 }
 div button: active {
    border: none;
 }
 div button: visited {
    border: none;
 }
button,input,optgroup,select,textarea{
    margin:0
}
button,input,select,textarea{
    background-color:transparent;
    color:inherit;
    font-size:inherit;
    line-height:inherit
}
a button :focus {
    outline: rgb(251,151,124) auto 1px;
}
button,input{
    overflow:visible
}
div button: focus {
    background-color: rgb(251,151,124);
    color: white;
    outline: none;      
}

 button: focus {
    outline: -webkit-focus-ring-color auto 1px;
    background-color: #f1f3f5;
    color: black;
}
li div button: focus {
    outline: -webkit-focus-ring-color auto 1px;
    background-color: #f1f3f5;
    color: black;
}
ul li button: focus {
    outline: -webkit-focus-ring-color auto 1px;
    background-color: #f1f3f5;
    color: black;
}
button,select{
    text-transform:none
}
#title:hover  span{
    display:inline-block;
}
[type=reset],[type=submit],button,html [type=button]{
    -webkit-appearance:button
}
[type=button]::-moz-focus-inner,[type=reset]::-moz-focus-inner,[type=submit]::-moz-focus-inner,button::-moz-focus-inner{
    border-style:none;
    padding:0
}
[type=button]:-moz-focusring,[type=reset]:-moz-focusring,[type=submit]:-moz-focusring,button:-moz-focusring{
    outline:1px dotted ButtonText
}
legend{
    box-sizing:border-box;
    color:inherit;
    display:table;
    max-width:100%;
    padding:0;
    white-space:normal
}
progress{
    display:inline-block;
    vertical-align:baseline
}
textarea{
    overflow:auto;
    resize:vertical
}
[type=checkbox],[type=radio]{
    box-sizing:border-box;
    padding:0
}
[type=number]::-webkit-inner-spin-button,[type=number]::-webkit-outer-spin-button{
    height:auto
}
[type=search]{
    -webkit-appearance:textfield;
    outline-offset:-2px
}
[type=search]::-webkit-search-cancel-button,[type=search]::-webkit-search-decoration{
    -webkit-appearance:none
}
::-webkit-file-upload-button{
    -webkit-appearance:button;
    font:inherit
}
details,menu{
    display:block
}
summary{
    display:list-item
}
canvas{
    display:inline-block
}
template{
    display:none
}
[tabindex],a,area,button,input,label,select,summary,textarea{
    -ms-touch-action:manipulation;
    touch-action:manipulation
}
[hidden]{
    display:none
}
[aria-busy=true]{
    cursor:progress
}
[aria-controls]{
    cursor:pointer
}
[aria-hidden=false][hidden]:not(:focus){
    clip:rect(0,0,0,0);
    display:inherit;
    position:absolute
}
[aria-disabled]{
    cursor:default
}
box-sizing: border-box;
 border: 0 solid #e5e7eb;
button, [role="button"] {
    cursor: pointer;
}
 li{
    list-style-type: none;
}`

export default createGlobalStyle`
    @font-face {
        font-family: 'Font Name';
        src: local('Font Name'), local('FontName'),
        url(../fonts/NewsGothicBTW01Roman) format('woff2'),
        url(../fonts/NewsGothicBTW01Roman2) format('woff');
        font-weight: 300;
        font-style: normal;
    }

    *,::after,::before{
        background-repeat:no-repeat;
        box-sizing:inherit
    }
    ::after,::before{
        text-decoration:inherit;
        vertical-align:inherit
    },
    :focus-visible {
        outline: none;
    }
    
    html{
        box-sizing:border-box;
        cursor:default;
        -ms-text-size-adjust:100%;
        -webkit-text-size-adjust:100%
    }
    article,aside,footer,header,nav,section{
        display:block
    }
    body{
        margin:0
    }
    h1{
        font-size:2em;
        margin:.67em 0
    }
    figcaption,figure,main{
        display:block
    }
    figure{
        margin:1em 40px
    }
    hr{
        box-sizing:content-box;
        height:0;
        overflow:visible
    }
    nav ol,nav ul{
        list-style:none
    }
    pre{
        font-family:monospace,monospace;
        font-size:1em
    }
    a{
        text-decoration:none;
        color:inherit;
        background-color:transparent;
        -webkit-text-decoration-skip:objects
    }
    abbr[title]{
        border-bottom:none;
        text-decoration:underline;
        text-decoration:underline dotted
    }
    b,strong{
        font-weight:inherit
    }
    b,strong{
        font-weight:bolder
    }
    code,kbd,samp{
        font-family:monospace,monospace;
        font-size:1em
    }
    dfn{
        font-style:italic
    }
    mark{
        background-color:#ff0;
        color:#000
    }
    small{
        font-size:80%
    }
    sub,sup{
        font-size:75%;
        line-height:0;
        position:relative;
        vertical-align:baseline
    }
    sub{
        bottom:-.25em
    }
    sup{
        top:-.5em
    }
    ::-moz-selection{
        background-color:#b3d4fc;
        color:#000;
        text-shadow:none
    }
    ::selection{
        background-color:#b3d4fc;
        color:#000;
        text-shadow:none
    }
    audio,canvas,iframe,img,svg,video{
        vertical-align:middle
    }
    audio,video{
        display:inline-block
    }
    audio:not([controls]){
        display:none;
        height:0
    }
    img{
        border-style:none
    }
    svg:not(:root){
        overflow:hidden
    }
    table{
        border-collapse:collapse
    }
    button,input,optgroup,select,textarea{
        margin:0
    }
    button,input,select,textarea{
        background-color:transparent;
        color:inherit;
        font-size:inherit;
        line-height:inherit
    }
    button,input{
        overflow:visible
    }
    button,select{
        text-transform:none
    }
    input[type="search"] {
        appearance: textfield;
        box-sizing: content-box;
    }
    [type=reset],[type=submit],button,html [type=button]{
        -webkit-appearance:button
    }
    [type=button]::-moz-focus-inner,[type=reset]::-moz-focus-inner,[type=submit]::-moz-focus-inner,button::-moz-focus-inner{
        border-style:none;
        padding:0
    }
    [type=button]:-moz-focusring,[type=reset]:-moz-focusring,[type=submit]:-moz-focusring,button:-moz-focusring{
        outline:1px dotted ButtonText
    }
    legend{
        box-sizing:border-box;
        color:inherit;
        display:table;
        max-width:100%;
        padding:0;
        white-space:normal
    }
    progress{
        display:inline-block;
        vertical-align:baseline
    }
    textarea{
        overflow:auto;
        resize:vertical
    }
    [type=checkbox],[type=radio]{
        box-sizing:border-box;
        padding:0
    }
    [type=number]::-webkit-inner-spin-button,[type=number]::-webkit-outer-spin-button{
        height:auto
    }
    [type=search]{
        -webkit-appearance:textfield;
        outline-offset:-2px
    }
    [type=search]::-webkit-search-cancel-button,[type=search]::-webkit-search-decoration{
        -webkit-appearance:none
    }
    ::-webkit-file-upload-button{
        -webkit-appearance:button;
        font:inherit
    }
    details,menu{
        display:block
    }
    summary{
        display:list-item
    }
    canvas{
        display:inline-block
    }
    template{
        display:none
    }
    [tabindex],a,area,button,input,label,select,summary,textarea{
        -ms-touch-action:manipulation;
        touch-action:manipulation
    }
    [hidden]{
        display:none
    }
    [aria-busy=true]{
        cursor:progress
    }
    [aria-controls]{
        cursor:pointer
    }
    [aria-hidden=false][hidden]:not(:focus){
        clip:rect(0,0,0,0);
        display:inherit;
        position:absolute
    }
    [aria-disabled]{
        cursor:default
    }
    box-sizing: border-box;
     border: 0 solid #e5e7eb;
    button, [role="button"] {
        cursor: pointer;
    }
     li{
        list-style-type: none;
    }
    
`;