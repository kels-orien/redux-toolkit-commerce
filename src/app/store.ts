import { configureStore, getDefaultMiddleware } from "@reduxjs/toolkit"
import { apiSlice } from "../features/products/products-api-slice"
import cartReducer  from "../features/cart/cart-slice"
import shippingReducer from "../features/shipping/shipping-slice"

export const store =  configureStore({
    reducer: {
        cart: cartReducer,
        shipping: shippingReducer,
        [apiSlice.reducerPath]: apiSlice.reducer
    },
    middleware: (getDefaultMiddleware) => {
        return getDefaultMiddleware().concat(apiSlice.middleware)
    }   
})

export type AppDispatch =  typeof store.dispatch
export type RootState = ReturnType<typeof store.getState>