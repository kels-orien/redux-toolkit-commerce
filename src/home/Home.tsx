import React, { useState } from 'react'
import { useFetchProductsQuery } from '../features/products/products-api-slice'
import styled, {css} from 'styled-components'
import Thumbnail from '../components/Thumbnail'
import Main from '../components/Main'
import Container from '../components/Container'
import Loading from '../components/Loading'
import rangeMap from '../utils/range-map'
import Skeleton from '../components/Skeleton'
import Wrapper from '../components/Wrapper'
import CardWrapper from '../components/CardWrapper'


function Home() {


  
  const { data = [], isFetching, isLoading } = useFetchProductsQuery();
  
  if (isLoading) return <Loading/>
  return (  
    <Main>
    <Container>
      {data.length > 0 ? (data.map((product) => (
        <Thumbnail key={product.id} product={product} />
      ))) : (<Wrapper>
                <CardWrapper>
                  {rangeMap(12, (i) => (
                     <Skeleton
                      key={i}
                      />
            ))}

        </CardWrapper>
      </Wrapper>)}
    </Container>
  </Main>
  )
}

export default Home
  
