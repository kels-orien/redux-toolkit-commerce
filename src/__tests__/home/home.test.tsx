import React from 'react'
import Home from '../../home'
import { BrowserRouter } from 'react-router-dom'
import {
    render,
    screen,
    within,
    fireEvent,
    RenderResult,
  } from '@testing-library/react';
import { Provider } from 'react-redux'
import { store } from '../../app/store'

const renderApp = (): RenderResult => 
render (
    <Provider store= { store }>
        <BrowserRouter>
        <Home/>
        </BrowserRouter>   
    </Provider>,
)
test('fetches and displays all products', async () => {
    const { asFragment, getByText } = renderApp()
    expect(asFragment()).toMatchSnapshot()
  
    // should show no user initially, and not be fetching a user
    //expect(getByTestId('data-values')).toHaveLength(20)
    const data = {title: 'Mens', contents: 'Your perfect pack for everyday use and walks in the forest. Stash your laptop (up to 15 inches) in the padded sleeve, your everyday'}
    
    const falseData = {
    title: 'Favorite Footware',
    contents: 'Flipflops are the best',
  }

  expect(await screen.findByText(data.title)).toBeInTheDocument()
  expect(screen.queryByText(falseData.title)).not.toBeInTheDocument()
})