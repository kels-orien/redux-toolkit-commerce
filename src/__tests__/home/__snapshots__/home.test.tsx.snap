// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetches and displays all products 1`] = `
<DocumentFragment>
  <div
    class="sc-iqAclL jeywhT"
  >
    <div
      class="sc-jrsJWt cdQzLn"
    >
      <div
        class="sc-kEqXSa iddVoK"
      />
    </div>
  </div>
</DocumentFragment>
`;
